features/breed_detail/src/test/java/com/example/catbreeds/breed_detail/unitTests/BreedDetailViewModelTest.kt

```diff
@@ -0,0 +1,177 @@
+        description = "A strong, moderately large cat.",
+        reference_image_id = "sibe",
+        isFavorite = isFav
+    )
+    val testBreed = getTestBreed("testId", false)
+}
+
+@ExperimentalCoroutinesApi
+class BreedDetailViewModelTest {
+
+    @get:Rule
+    val instantTaskExecutorRule = InstantTaskExecutorRule()
+    @get:Rule
+    val mainDispatcherRule = MainDispatcherRule()
+    @get:Rule
+    val mockkRule = MockKRule(this)
+
+    @RelaxedMockK
+    private lateinit var breedRepository: BreedRepository
+
+    private lateinit var savedStateHandle: SavedStateHandle
+
+    private val vmUnderTest: BreedDetailViewModel by lazy {
+        spyk(
+            BreedDetailViewModel(
+                breedRepository = breedRepository,
+                savedStateHandle = savedStateHandle
+            ),
+        )
+    }
+
+    @Before
+    fun setup() {
+        savedStateHandle = SavedStateHandle()
+    }
+
+    @Test
+    fun `WHEN initialized with valid ID SHOULD load breed`() = runTest {
+        // GIVEN
+        val targetBreed = TestBreedDetailData.testBreed
+        val breedId = targetBreed.id
+        savedStateHandle["breedId"] = breedId
+        coEvery { breedRepository.getBreedById(breedId) } returns targetBreed
+        every { breedRepository.getFavoriteBreeds() } returns flowOf(emptyList())
+
+        // WHEN
+        val vm = vmUnderTest
+        advanceUntilIdle()
+
+        // THEN
+        assertEquals(targetBreed.copy(isFavorite = false), vm.breed.value)
+    }
+
+    @Test
+    fun `WHEN initialized with invalid ID SHOULD handle not found by setting breed to null`() = runTest {
+        // GIVEN
+        val nonExistentId = "non_existent"
+        savedStateHandle["breedId"] = nonExistentId
+        coEvery { breedRepository.getBreedById(nonExistentId) } returns null
+        every { breedRepository.getFavoriteBreeds() } returns flowOf(emptyList())
+
+        // WHEN
+        val vm = vmUnderTest
+        advanceUntilIdle()
+
+        // THEN
+        assertNull(vm.breed.value)
+    }
+
+    @Test
+    fun `WHEN favorite status updates from flow SHOULD update the ViewModel's breed favorite status`() = runTest {
+        // GIVEN
+        val targetBreed = TestBreedDetailData.testBreed
+        val breedId = targetBreed.id
+        savedStateHandle["breedId"] = breedId
+        val favoriteFlow = MutableStateFlow(emptyList<Breed>())
+        coEvery { breedRepository.getBreedById(breedId) } returns targetBreed
+        every { breedRepository.getFavoriteBreeds() } returns favoriteFlow
+        val vm = vmUnderTest
+        advanceUntilIdle()
+        assertFalse(vm.breed.value?.isFavorite ?: true)
+
+        // WHEN
+        favoriteFlow.value = listOf(targetBreed.copy(isFavorite = true))
+        advanceUntilIdle()
+
+        // THEN
+        assertTrue(vm.breed.value?.isFavorite ?: false)
+    }
+
+    @Test
+    fun `WHEN toggling favorite SHOULD update state and call repository correctly`() = runTest {
```

Comment:
we usually put the "ands" in uppercase 

```suggestion
    fun `WHEN toggling favorite SHOULD update state AND call repository correctly`() = runTest {
```

Reply:
Thanks, I wasn't aware